students/rhamersky/Lesson_5/mailroom_part3.py

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python3
+# Description: This program will allow a user to input a donor and a donation.
+# The user will also be able to send out thank you notes to the donors.
+# Developer: Ryan Hamersky
+# Date: 05/13/2018
+# Rev: A - 05/16/2018 updated code to reflect comments made by the instructor in mailroom_part.py.
+#      Update variables to snake_case. Removed unnecessary 'continues' and 'returns'.
+#      Changed 'bool' variable to 'bool_flag'. Updated misleading function names.
+#      Reduced the amount of unnecessary print() statements.
+
+# -----Data Section-----
+import os
+str_name = None     # --> User's name
+int_option = int()  # --> User's option.
+tup_donor_names = (["William Gates, III",1,10000.0],["Mark Zuckerberg",1,20000.0],
+                   ["Jeff Bezos",1,1200.0],["Paul Allen",1,5000.0],
+                   ["Mike Scott",1,25.0])  # --> Tuple that stores all donor's information
+str_salutation = ""   # --> Email salutation
+str_body = ""         # --> Email body
+str_valediction = ""  # --> Email valediction
+
+
+# -----Process Section-----
+#  Methods defined in the menu class are related to general menu checks or options based on user inputs.
+
+class menu(object):
+    @staticmethod
+    def menu_options():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns int data (1-4) which is used to determine what menu option will be executed.
+        '''
+        while True:
+            try:
+                int_choice = int(input(str_name + ", please select an option above. "))
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+            # Verify that the user selected an option 1 through option 4.
+            else:
+                if 1 <= int_choice <= 4:
```

Comment:
you can check if `int_choice in menu_dispatch`

Reply:
Thanks for the tip. This line I always had to remember to update if I changed the number of options in the menu. This will take care of it since I am using the dictionary dispatch to go into my functions.